packages/console/src/pages/ApplicationDetails/index.tsx

```diff
@@ -27,45 +29,118 @@ type OidcConfig = {
     reset(data);
   }, [data, reset]);
 
+  const {
+    field: { value: redirectUris, onChange: onRedirectUriChange },
+  } = useController({
+    control,
+    name: 'oidcClientMetadata.redirectUris',
+    defaultValue: [],
+  });
+
+  const {
+    field: { value: postSignOutRedirectUris, onChange: onPostSignOutRedirectUriChange },
+  } = useController({
+    control,
+    name: 'oidcClientMetadata.postLogoutRedirectUris',
+    defaultValue: [],
+  });
+
   const onSubmit = handleSubmit((formData) => {
     console.log(formData);
   });
 
+  const isAdvancedSettings = location.pathname.includes('advanced-settings');
+
+  const SettingsPage = useMemo(() => {
+    return (
+      oidcConfig && (
+        <>
+          <FormField isRequired title="admin_console.application_details.application_name">
+            <TextInput {...register('name', { required: true })} />
+          </FormField>
+          <FormField title="admin_console.application_details.description">
+            <TextInput {...register('description')} />
+          </FormField>
+          <FormField title="admin_console.application_details.authorization_endpoint">
+            <CopyToClipboard
+              className={styles.textField}
+              value={oidcConfig.authorization_endpoint}
+            />
+          </FormField>
+          <FormField title="admin_console.application_details.redirect_uri">
+            <MultilineInput
+              value={redirectUris}
+              onChange={(value) => {
+                onRedirectUriChange(value);
+              }}
+            />
+          </FormField>
+          <FormField title="admin_console.application_details.post_sign_out_redirect_uri">
+            <MultilineInput
+              value={postSignOutRedirectUris}
+              onChange={(value) => {
+                onPostSignOutRedirectUriChange(value);
+              }}
+            />
+          </FormField>
+        </>
+      )
+    );
+  }, [
+    oidcConfig,
+    onPostSignOutRedirectUriChange,
+    onRedirectUriChange,
+    postSignOutRedirectUris,
+    redirectUris,
+    register,
+  ]);
+
+  const AdvancedSettingsPage = useMemo(() => {
+    return (
+      oidcConfig && (
+        <>
+          <FormField title="admin_console.application_details.token_endpoint">
+            <CopyToClipboard className={styles.textField} value={oidcConfig.token_endpoint} />
+          </FormField>
+          <FormField title="admin_console.application_details.user_info_endpoint">
+            <CopyToClipboard className={styles.textField} value={oidcConfig.userinfo_endpoint} />
+          </FormField>
+        </>
+      )
+    );
+  }, [oidcConfig]);
+
   return (
     <div className={styles.container}>
       <BackLink to="/applications">{t('application_details.back_to_applications')}</BackLink>
       {isLoading && <div>loading</div>}
       {error && <div>{`error occurred: ${error.metadata.code}`}</div>}
-      {dataFetched && (
+      {data && oidcConfig && (
         <>
```

Comment:
no need for the empty fragment? as <Card> is the only element inside?

Reply:
No. There are two `Card` inside.